app/models/shipit/task.rb

```diff
@@ -382,6 +386,18 @@ def self.zombies
         .reject(&:alive?)
     end
 
+    def retry_if_necessary
+      return if max_retries.nil?
+
+      if retry_attempt < max_retries
+        retry_task = duplicate_task
+        retry_task.retry_attempt = duplicate_task.retry_attempt + 1
+        retry_task.save!
+
+        retry_task.enqueue
```

Comment:
There is also a good argument here to run right away instead of enqueueing, so we can get the final state of a task (even if all retries time out) and take appropriate action without having to wait on the job to be enqueued. 

We ultimately elected enqueue because:
- same reason as using enqueue for tasks right now: allow tasks to be performed asynchronously
- the app does not process a large number of jobs, so there likely won't be a big delay in getting the final result of the task as chances are the job will be enqueued in the very near future

Would love to hear others thoughts on this

Reply:
Operation wise it's better to re-enqueue. Very long running jobs make shipit slow to deploy.